pyramid_swagger/api.py

```diff
@@ -266,23 +266,49 @@ def _get_target_url(spec, target, current_path=''):
         )
 
 
+def _ensure_sane_key(fragment, mar_target):
+    '''
+    This function patches a nasty bug with the
+    'allOff' keyword in a spec, our resolved
```

Comment:
What about this relies on allOf? Is it an issue with a number of nested keys like allOf and we're just noticing now? Does it happen with things like `additionalProperties` or `items`?